Latest version 1.3 crashes with refine_corner flag

### System information (version)

* Operating System / Platform => Ubuntu 22
* OpenCV =>  4.5.5
* Ceres => 2.0
* Boost =>  1.74
* C++ => 17
* Compiler =>  g++11.4

### Vision system

* Number of cameras => 6
* Types of cameras =>  fisheye, 5mm
* Multicamera configurations => non-overlapping
* Configuration file => *.yml

0000001 | 2024-10-06, 19:44:21.266054 [info] - Nb of cameras : 6   Nb of Boards : 9   Refined Corners : true   Distortion mode : 1
0000002 | 2024-10-06, 19:44:21.266414 [info] - Extraction camera 001
0000003 | 2024-10-06, 19:44:21.270420 [info] - Number of threads for board detection :: 8
0001500 | 2024-10-06, 19:44:22.039815 [info] - Number of detected corners :: 16
terminate called after throwing an instance of 'cv::Exception'
  what():  OpenCV(4.5.5) /home/vsw/opencv/opencv/modules/imgproc/src/filterengine.hpp:367: error: (-215:Assertion failed) anchor.inside(Rect(0, 0, ksize.width, ksize.height)) in function 'normalizeAnchor'

Also after boards extraction i got this:

0011632 | 2024-10-07, 08:51:40.934021 [info] - Number of detected corners :: 16
0011633 | 2024-10-07, 08:51:41.724832 [info] - Number of detected corners :: 16
0011634 | 2024-10-07, 08:51:42.597949 [info] - Board extraction done!
0011635 | 2024-10-07, 08:51:42.597994 [info] - Intrinsic calibration initiated
0011636 | 2024-10-07, 08:51:42.598002 [info] - Initializing camera calibration using images
0011637 | 2024-10-07, 08:51:42.598008 [info] - NB of board available in this camera :: 419
0011638 | 2024-10-07, 08:51:42.598023 [info] - NB of frames where this camera saw a board :: 291
0011639 | 2024-10-07, 08:51:44.884207 [info] - cameraMatrix : [779.2226013779195, 0, 1223.5;
 0, 779.2226013779195, 1023.5;
 0, 0, 1]
0011640 | 2024-10-07, 08:51:44.884275 [info] - distCoeffs : [0;
 0;
 0;
 0]
0011641 | 2024-10-07, 08:51:44.884312 [info] - NB of board available in this camera :: 899
0011642 | 2024-10-07, 08:51:44.884318 [info] - NB of frames where this camera saw a board :: 499
terminate called after throwing an instance of 'cv::Exception'
  what():  OpenCV(4.5.5) /home/vsw/opencv/opencv/modules/calib3d/src/fisheye.cpp:1400: error: (-215:Assertion failed) fabs(norm_u1) > 0 in function 'InitExtrinsics'

Aborted (core dumped)

Here i changed to Brown lens model

0001916 | 2024-10-07, 09:51:24.069684 [info] - Number of detected corners :: 16
0001917 | 2024-10-07, 09:51:25.190587 [info] - Number of detected corners :: 16
0001918 | 2024-10-07, 09:51:25.444131 [info] - Number of detected corners :: 16
0001919 | 2024-10-07, 09:51:26.623468 [info] - Board extraction done!
0001920 | 2024-10-07, 09:51:26.623509 [info] - Intrinsic calibration initiated
0001921 | 2024-10-07, 09:51:26.623517 [info] - Initializing camera calibration using images
0001922 | 2024-10-07, 09:51:26.623523 [info] - NB of board available in this camera :: 419
0001923 | 2024-10-07, 09:51:26.623529 [info] - NB of frames where this camera saw a board :: 291
terminate called after throwing an instance of 'cv::Exception'
  what():  OpenCV(4.5.5) /home/vsw/opencv/opencv/modules/calib3d/src/calibration.cpp:1401: error: (-211:One of the arguments' values is out of range) image width and height must be positive in function 'cvCalibrateCamera2Internal'

Aborted (core dumped)

I would most likely use the fisheye model for this camera. When you use the fisheye model, are you facing this issue?
```
terminate called after throwing an instance of 'cv::Exception'
what(): OpenCV(4.5.5) /home/vsw/opencv/opencv/modules/calib3d/src/fisheye.cpp:1400: error: (-215:Assertion failed) fabs(norm_u1) > 0 in function 'InitExtrinsics'
```
We have addressed a similar problem in a previous commit to avoid points outside of the image. Maybe you can adjust the tolerance threshold? I hope it will help [Link](https://github.com/rameau-fr/MC-Calib/blob/a999bdcd89d69168f1f7c7820bb26452023057db/McCalib/include/Camera.hpp#L30)

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -17,8 +17,8 @@ on:
   workflow_dispatch:
 
 env:
-  MC_CALIB_PROD_DOCKER_IMG: "bailool/mc-calib-prod:latest"
-  MC_CALIB_DEV_DOCKER_IMG: "bailool/mc-calib-dev:latest"
+  MC_CALIB_PROD_DOCKER_IMG: bailool/mc-calib-prod:latest
+  MC_CALIB_DEV_DOCKER_IMG: bailool/mc-calib-dev:latest
 
 # A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
@@ -46,7 +46,7 @@ jobs:
       - name: Run cppcheck
         uses: addnab/docker-run-action@v3
         with:
-          image: $MC_CALIB_DEV_DOCKER_IMG
+          image: ${{env.MC_CALIB_DEV_DOCKER_IMG}}
           options: -v ${{ github.workspace }}:/home/MC-Calib
           run: |
             cppcheck MC-Calib/McCalib/include
@@ -62,7 +62,7 @@ jobs:
       - name: Run clang-tidy
         uses: addnab/docker-run-action@v3
         with:
-          image: $MC_CALIB_DEV_DOCKER_IMG
+          image: ${{env.MC_CALIB_DEV_DOCKER_IMG}}
           options: -v ${{ github.workspace }}:/home/MC-Calib
           run: |
             mkdir MC-Calib/build && cd MC-Calib/build
@@ -94,7 +94,7 @@ jobs:
       - name: Run tests
         uses: addnab/docker-run-action@v3
         with:
-          image: $MC_CALIB_DEV_DOCKER_IMG
+          image: ${{env.MC_CALIB_DEV_DOCKER_IMG}}
           options: -v ${{ github.workspace }}:/home/MC-Calib
           run: |
             mkdir MC-Calib/build && cd MC-Calib/build
@@ -126,7 +126,7 @@ jobs:
       - name: Run tests
         uses: addnab/docker-run-action@v3
         with:
-          image: $MC_CALIB_DEV_DOCKER_IMG
+          image: ${{env.MC_CALIB_DEV_DOCKER_IMG}}
           options: -v ${{ github.workspace }}:/home/MC-Calib
           run: |
             mkdir MC-Calib/build && cd MC-Calib/build
@@ -169,7 +169,7 @@ jobs:
       - name: Run tests
         uses: addnab/docker-run-action@v3
         with:
-          image: $MC_CALIB_PROD_DOCKER_IMG
+          image: ${{env.MC_CALIB_PROD_DOCKER_IMG}}
           options: -v ${{ github.workspace }}:/home/MC-Calib:rw
           run: |
             mkdir MC-Calib/build && cd MC-Calib/build
@@ -232,7 +232,7 @@ jobs:
       - name: Run tests
         uses: addnab/docker-run-action@v3
         with:
-          image: $MC_CALIB_DEV_DOCKER_IMG
+          image: ${{env.MC_CALIB_DEV_DOCKER_IMG}}
           options: -v ${{ github.workspace }}:${{ github.workspace }}
           run: |
             mkdir ${{ github.workspace }}/build && cd ${{ github.workspace }}/build
@@ -270,7 +270,7 @@ jobs:
       - name: Run python linters
         uses: addnab/docker-run-action@v3
         with:
-          image: $MC_CALIB_DEV_DOCKER_IMG
+          image: ${{env.MC_CALIB_DEV_DOCKER_IMG}}
           options: -v ${{ github.workspace }}:/home/MC-Calib
           run: |
             cd MC-Calib/python_utils
@@ -296,7 +296,7 @@ jobs:
       - name: Run python utils
         uses: addnab/docker-run-action@v3
         with:
-          image: $MC_CALIB_PROD_DOCKER_IMG
+          image: ${{env.MC_CALIB_PROD_DOCKER_IMG}}
           options: -v ${{ github.workspace }}:/home/MC-Calib:rw
           run: |
             cd MC-Calib/python_utils

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -17,8 +17,8 @@ on:
   workflow_dispatch:
 
 env:
-  MC_CALIB_PROD_DOCKER_IMG: bailool/mc-calib-prod:opencv455
-  MC_CALIB_DEV_DOCKER_IMG: bailool/mc-calib-dev:opencv455
+  MC_CALIB_PROD_DOCKER_IMG: bailool/mc-calib-prod:opencv420
+  MC_CALIB_DEV_DOCKER_IMG: bailool/mc-calib-dev:opencv420
 
 # A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -17,8 +17,8 @@ on:
   workflow_dispatch:
 
 env:
-  MC_CALIB_PROD_DOCKER_IMG: bailool/mc-calib-prod:latest
-  MC_CALIB_DEV_DOCKER_IMG: bailool/mc-calib-dev:latest
+  MC_CALIB_PROD_DOCKER_IMG: bailool/mc-calib-prod:opencv455
+  MC_CALIB_DEV_DOCKER_IMG: bailool/mc-calib-dev:opencv455
 
 # A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -16,6 +16,10 @@ on:
   # Allows you to run this workflow manually from the Actions tab
   workflow_dispatch:
 
+env:
+  MC_CALIB_PROD_DOCKER_IMG: bailool/mc-calib-prod:latest
+  MC_CALIB_DEV_DOCKER_IMG: bailool/mc-calib-dev:latest
+
 # A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
   clang-format-lint:
@@ -42,7 +46,7 @@ jobs:
       - name: Run cppcheck
         uses: addnab/docker-run-action@v3
         with:
-          image: bailool/mc-calib-dev:latest
+          image: $MC_CALIB_DEV_DOCKER_IMG
           options: -v ${{ github.workspace }}:/home/MC-Calib
           run: |
             cppcheck MC-Calib/McCalib/include
@@ -58,7 +62,7 @@ jobs:
       - name: Run clang-tidy
         uses: addnab/docker-run-action@v3
         with:
-          image: bailool/mc-calib-dev:latest
+          image: $MC_CALIB_DEV_DOCKER_IMG
           options: -v ${{ github.workspace }}:/home/MC-Calib
           run: |
             mkdir MC-Calib/build && cd MC-Calib/build
@@ -90,7 +94,7 @@ jobs:
       - name: Run tests
         uses: addnab/docker-run-action@v3
         with:
-          image: bailool/mc-calib-dev:latest
+          image: $MC_CALIB_DEV_DOCKER_IMG
           options: -v ${{ github.workspace }}:/home/MC-Calib
           run: |
             mkdir MC-Calib/build && cd MC-Calib/build
@@ -122,7 +126,7 @@ jobs:
       - name: Run tests
         uses: addnab/docker-run-action@v3
         with:
-          image: bailool/mc-calib-dev:latest
+          image: $MC_CALIB_DEV_DOCKER_IMG
           options: -v ${{ github.workspace }}:/home/MC-Calib
           run: |
             mkdir MC-Calib/build && cd MC-Calib/build
@@ -165,7 +169,7 @@ jobs:
       - name: Run tests
         uses: addnab/docker-run-action@v3
         with:
-          image: bailool/mc-calib-prod:latest
+          image: $MC_CALIB_PROD_DOCKER_IMG
           options: -v ${{ github.workspace }}:/home/MC-Calib:rw
           run: |
             mkdir MC-Calib/build && cd MC-Calib/build
@@ -228,7 +232,7 @@ jobs:
       - name: Run tests
         uses: addnab/docker-run-action@v3
         with:
-          image: bailool/mc-calib-dev:latest
+          image: $MC_CALIB_DEV_DOCKER_IMG
           options: -v ${{ github.workspace }}:${{ github.workspace }}
           run: |
             mkdir ${{ github.workspace }}/build && cd ${{ github.workspace }}/build
@@ -266,7 +270,7 @@ jobs:
       - name: Run python linters
         uses: addnab/docker-run-action@v3
         with:
-          image: bailool/mc-calib-dev:latest
+          image: $MC_CALIB_DEV_DOCKER_IMG
           options: -v ${{ github.workspace }}:/home/MC-Calib
           run: |
             cd MC-Calib/python_utils
@@ -292,7 +296,7 @@ jobs:
       - name: Run python utils
         uses: addnab/docker-run-action@v3
         with:
-          image: bailool/mc-calib-prod:latest
+          image: $MC_CALIB_PROD_DOCKER_IMG
           options: -v ${{ github.workspace }}:/home/MC-Calib:rw
           run: |
             cd MC-Calib/python_utils

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -17,8 +17,8 @@ on:
   workflow_dispatch:
 
 env:
-  MC_CALIB_PROD_DOCKER_IMG: bailool/mc-calib-prod:latest
-  MC_CALIB_DEV_DOCKER_IMG: bailool/mc-calib-dev:latest
+  MC_CALIB_PROD_DOCKER_IMG: "bailool/mc-calib-prod:latest"
+  MC_CALIB_DEV_DOCKER_IMG: "bailool/mc-calib-dev:latest"
 
 # A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,9 +1,9 @@
-FROM ubuntu:20.04 as prod
+FROM ubuntu:22.04 as prod
 
 ENV DEBIAN_FRONTEND noninteractive
 RUN apt update && apt install -y --no-install-recommends apt-utils && \
 	apt upgrade -y && apt autoremove -y && \
-	apt install -y build-essential cmake && \
+	apt install -y build-essential cmake wget unzip git && \
 	apt install -y python-is-python3 python3-pip && \
 	rm -rf /var/lib/apt/lists/*
 
@@ -15,7 +15,14 @@ WORKDIR /home
 #------------------------------	#
 #     INSTALL OPENCV 4		    #
 #------------------------------	#
-RUN apt update && apt install -y libopencv-dev && \
+RUN wget -O opencv.zip https://github.com/opencv/opencv/archive/4.5.5.zip && \
+	wget -O opencv_contrib.zip https://github.com/opencv/opencv_contrib/archive/4.5.5.zip && \
+	unzip opencv.zip && unzip opencv_contrib.zip && \
+	mkdir -p build && cd build && \
+	cmake -DOPENCV_EXTRA_MODULES_PATH=../opencv_contrib-4.5.5/modules ../opencv-4.5.5 && \
+	cmake --build . --target install && \
+	cd /home && rm opencv.zip && rm opencv_contrib.zip && \ 
+	rm -rf opencv-4.5.5 && rm -rf opencv_contrib-4.5.5 && rm -rf build && \
 	rm -rf /var/lib/apt/lists/*
 
 #------------------------------	#
@@ -27,36 +34,36 @@ RUN apt update && apt install -y libboost-all-dev && \
 #------------------------------	#
 #     Install Ceres		        #
 #------------------------------	#
-RUN apt update && \
-	apt install -y libgoogle-glog-dev libgflags-dev libatlas-base-dev libeigen3-dev libsuitesparse-dev && \
+RUN apt update && apt install -y libgoogle-glog-dev libgflags-dev libatlas-base-dev libeigen3-dev libsuitesparse-dev && \
 	rm -rf /var/lib/apt/lists/*
-RUN apt update && apt install -y git && \
-	/bin/bash -c "git clone https://ceres-solver.googlesource.com/ceres-solver && \
+RUN git clone --branch 20240722.0 --single-branch https://github.com/abseil/abseil-cpp.git && \
+	cd abseil-cpp && mkdir build && cd build && \
+	cmake .. && cmake --build . --target install && \
+	cd /home && rm -rf abseil-cpp && \
+	rm -rf /var/lib/apt/lists/*
+RUN git clone --branch 2.2.0 --single-branch https://github.com/ceres-solver/ceres-solver.git && \
 	cd ceres-solver && mkdir build && cd build && cmake .. && \
 	make && make install && \
-	cd /home && rm -rf ceres-solver" && \
+	cd /home && rm -rf ceres-solver && \
 	rm -rf /var/lib/apt/lists/*
 
 # Install python requirements for python_utils scripts
 RUN --mount=type=bind,source=python_utils/requirements_prod.txt,target=/tmp/requirements.txt \
-    pip install --requirement /tmp/requirements.txt && \
+    python -m pip install --requirement /tmp/requirements.txt && \
 	rm -rf /var/lib/apt/lists/*
 
 FROM prod as dev
-
-RUN apt update && apt install -y python3-opencv && \
-	rm -rf /var/lib/apt/lists/*
 
 #------------------------------	#
 #     Doxygen				    #
 #------------------------------	#
 RUN apt update && apt install -y flex bison && \ 
-	/bin/bash -c "git clone https://github.com/doxygen/doxygen.git && \
+	git clone https://github.com/doxygen/doxygen.git && \
 	cd doxygen && mkdir build && cd build &&\
 	cmake -G 'Unix Makefiles' .. && \
 	make && \ 
 	make install && \
-	cd /home && rm -rf doxygen" && \
+	cd /home && rm -rf doxygen && \
 	rm -rf /var/lib/apt/lists/*
 
 #------------------------------	#
@@ -67,5 +74,5 @@ RUN apt update && apt install -y cppcheck clang-tidy valgrind lcov && \
 
 # Install python requirements for python_utils scripts
 RUN --mount=type=bind,source=python_utils/requirements_dev.txt,target=/tmp/requirements.txt \
-    pip install --requirement /tmp/requirements.txt && \
+    python -m pip install --requirement /tmp/requirements.txt && \
 	rm -rf /var/lib/apt/lists/*

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,4 +1,4 @@
-FROM ubuntu:22.04 as prod
+FROM ubuntu:20.04 as prod
 
 ENV DEBIAN_FRONTEND noninteractive
 RUN apt update && apt install -y --no-install-recommends apt-utils && \
@@ -15,14 +15,14 @@ WORKDIR /home
 #------------------------------	#
 #     INSTALL OPENCV 4		    #
 #------------------------------	#
-RUN wget -O opencv.zip https://github.com/opencv/opencv/archive/4.5.5.zip && \
-	wget -O opencv_contrib.zip https://github.com/opencv/opencv_contrib/archive/4.5.5.zip && \
+RUN wget -O opencv.zip https://github.com/opencv/opencv/archive/4.2.0.zip && \
+	wget -O opencv_contrib.zip https://github.com/opencv/opencv_contrib/archive/4.2.0.zip && \
 	unzip opencv.zip && unzip opencv_contrib.zip && \
 	mkdir -p build && cd build && \
-	cmake -DOPENCV_EXTRA_MODULES_PATH=../opencv_contrib-4.5.5/modules ../opencv-4.5.5 && \
-	cmake --build . --target install && \
+	cmake -DOPENCV_EXTRA_MODULES_PATH=../opencv_contrib-4.2.0/modules ../opencv-4.2.0 && \
+	cmake --build . --target install -- -j4 && \
 	cd /home && rm opencv.zip && rm opencv_contrib.zip && \ 
-	rm -rf opencv-4.5.5 && rm -rf opencv_contrib-4.5.5 && rm -rf build && \
+	rm -rf opencv-4.2.0 && rm -rf opencv_contrib-4.2.0 && rm -rf build && \
 	rm -rf /var/lib/apt/lists/*
 
 #------------------------------	#
@@ -38,17 +38,18 @@ RUN apt update && apt install -y libgoogle-glog-dev libgflags-dev libatlas-base-
 	rm -rf /var/lib/apt/lists/*
 RUN git clone --branch 20240722.0 --single-branch https://github.com/abseil/abseil-cpp.git && \
 	cd abseil-cpp && mkdir build && cd build && \
-	cmake .. && cmake --build . --target install && \
+	cmake .. && cmake --build . --target install -- -j4 && \
 	cd /home && rm -rf abseil-cpp && \
 	rm -rf /var/lib/apt/lists/*
 RUN git clone --branch 2.2.0 --single-branch https://github.com/ceres-solver/ceres-solver.git && \
 	cd ceres-solver && mkdir build && cd build && cmake .. && \
-	make && make install && \
+	make -j4 && make -j4 install && \
 	cd /home && rm -rf ceres-solver && \
 	rm -rf /var/lib/apt/lists/*
 
 # Install python requirements for python_utils scripts
 RUN --mount=type=bind,source=python_utils/requirements_prod.txt,target=/tmp/requirements.txt \
+    apt update && apt install -y libgl1 && \
     python -m pip install --requirement /tmp/requirements.txt && \
 	rm -rf /var/lib/apt/lists/*
 
@@ -61,8 +62,8 @@ RUN apt update && apt install -y flex bison && \
 	git clone https://github.com/doxygen/doxygen.git && \
 	cd doxygen && mkdir build && cd build &&\
 	cmake -G 'Unix Makefiles' .. && \
-	make && \ 
-	make install && \
+	make -j4 && \ 
+	make -j4 install && \
 	cd /home && rm -rf doxygen && \
 	rm -rf /var/lib/apt/lists/*
 

--- a/README.md
+++ b/README.md
@@ -7,7 +7,7 @@ Toolbox described in the paper ["MC-Calib: A generic and robust calibration tool
 
 # Installation
 
-Requirements: Ceres, Boost, OpenCV {4.2.x, 4.5.x}, c++17 
+Requirements: Ceres, Boost, OpenCV 4.5.5, c++17 
 
 For Windows users, follow [this installation guide](/docs/Windows.md)
 
@@ -38,7 +38,7 @@ There are several ways to get the environment ready. Choose any of them:
 
 3. Alternatively, every dependency can be installed independently without docker:
 
-   - [Install](https://docs.opencv.org/4.5.2/d7/d9f/tutorial_linux_install.html) OpenCV 4.5.x. Either instal system-wide with `sudo make install` or link to your `build` in `CmakeLists.txt`.
+   - [Install](https://docs.opencv.org/4.5.5/d7/d9f/tutorial_linux_install.html) OpenCV 4.5.5. Either instal system-wide with `sudo make install` or link to your `build` in `CmakeLists.txt`.
 
    - Follow [installation guidelines](http://ceres-solver.org/installation.html#linux) to install Ceres.
 

--- a/README.md
+++ b/README.md
@@ -20,8 +20,8 @@ There are several ways to get the environment ready. Choose any of them:
    - Pull the image:
 
      ```bash
-     docker pull bailool/mc-calib-prod # production environment
-     docker pull bailool/mc-calib-dev  # development environment
+     docker pull bailool/mc-calib-prod:opencv455 # production environment
+     docker pull bailool/mc-calib-dev:opencv455  # development environment
      ```
 
    - Run pulled image (set `PATH_TO_REPO_ROOT` and `PATH_TO_DATA` appropriately):
@@ -31,7 +31,7 @@ There are several ways to get the environment ready. Choose any of them:
                   -ti --rm \
                   --volume="$PATH_TO_REPO_ROOT:/home/MC-Calib" \
                   --volume="$PATH_TO_DATA:/home/MC-Calib/data" \
-                  bailool/mc-calib-prod
+                  bailool/mc-calib-prod:opencv455
       ```
       
 2. It is also possible to build the docker environment manually (see [instructions](/docs/Docker.md))

--- a/README.md
+++ b/README.md
@@ -7,7 +7,7 @@ Toolbox described in the paper ["MC-Calib: A generic and robust calibration tool
 
 # Installation
 
-Requirements: Ceres, Boost, OpenCV 4.5.5, c++17 
+Requirements: Ceres, Boost, OpenCV 4.2.0, c++17 
 
 For Windows users, follow [this installation guide](/docs/Windows.md)
 
@@ -20,8 +20,8 @@ There are several ways to get the environment ready. Choose any of them:
    - Pull the image:
 
      ```bash
-     docker pull bailool/mc-calib-prod:opencv455 # production environment
-     docker pull bailool/mc-calib-dev:opencv455  # development environment
+     docker pull bailool/mc-calib-prod:opencv420 # production environment
+     docker pull bailool/mc-calib-dev:opencv420  # development environment
      ```
 
    - Run pulled image (set `PATH_TO_REPO_ROOT` and `PATH_TO_DATA` appropriately):
@@ -31,14 +31,14 @@ There are several ways to get the environment ready. Choose any of them:
                   -ti --rm \
                   --volume="$PATH_TO_REPO_ROOT:/home/MC-Calib" \
                   --volume="$PATH_TO_DATA:/home/MC-Calib/data" \
-                  bailool/mc-calib-prod:opencv455
+                  bailool/mc-calib-prod:opencv420
       ```
       
 2. It is also possible to build the docker environment manually (see [instructions](/docs/Docker.md))
 
 3. Alternatively, every dependency can be installed independently without docker:
 
-   - [Install](https://docs.opencv.org/4.5.5/d7/d9f/tutorial_linux_install.html) OpenCV 4.5.5. Either instal system-wide with `sudo make install` or link to your `build` in `CmakeLists.txt`.
+   - [Install](https://docs.opencv.org/4.2.0/d7/d9f/tutorial_linux_install.html) OpenCV 4.2.0. Either instal system-wide with `sudo make install` or link to your `build` in `CmakeLists.txt`.
 
    - Follow [installation guidelines](http://ceres-solver.org/installation.html#linux) to install Ceres.
 

